nav2_map_server/src/vo_server/vector_object_shapes.cpp

```diff
@@ -0,0 +1,588 @@
+// Copyright (c) 2023 Samsung R&D Institute Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_map_server/vector_object_shapes.hpp"
+
+#include <uuid/uuid.h>
+#include <math.h>
+#include <cmath>
+#include <limits>
+#include <stdexcept>
+
+#include "geometry_msgs/msg/pose_stamped.hpp"
+
+#include "nav2_util/occ_grid_utils.hpp"
+#include "nav2_util/occ_grid_values.hpp"
+#include "nav2_util/raytrace_line_2d.hpp"
+#include "nav2_util/robot_utils.hpp"
+
+namespace nav2_map_server
+{
+
+// ---------- Shape ----------
+
+Shape::Shape(const nav2_util::LifecycleNode::WeakPtr & node)
+: type_(UNKNOWN), node_(node)
+{}
+
+Shape::~Shape()
+{}
+
+ShapeType Shape::getType()
+{
+  return type_;
+}
+
+bool Shape::obtainShapeUUID(const std::string & shape_name, unsigned char * out_uuid)
+{
+  auto node = node_.lock();
+  if (!node) {
+    throw std::runtime_error{"Failed to lock node"};
+  }
+
+  try {
+    // Try to get shape UUID from ROS-parameters
+    std::string uuid_str = getParameter(
+      node, shape_name + ".uuid", rclcpp::PARAMETER_STRING).as_string();
+    if (uuid_parse(uuid_str.c_str(), out_uuid)) {
```

Comment:
if false? Does this return an int or something?

Reply:
`uuid_parse()` returns 0 in normal case and non-zero in case of failure (according to documentation, failure value is -1). Switched to comparison with 0 in https://github.com/ros-planning/navigation2/pull/3930/commits/43af6df733237b6f5ef8c91d0757b2ceda43a2de